pkg/storage/calico/caliconodestatus_storage.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2019 Tigera, Inc. All rights reserved.
+
+package calico
+
+import (
+	"reflect"
+
+	"golang.org/x/net/context"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/generic/registry"
+	"k8s.io/apiserver/pkg/storage"
+	etcd "k8s.io/apiserver/pkg/storage/etcd3"
+	"k8s.io/apiserver/pkg/storage/storagebackend/factory"
+
+	api "github.com/projectcalico/api/pkg/apis/projectcalico/v3"
+	"github.com/projectcalico/libcalico-go/lib/clientv3"
+	"github.com/projectcalico/libcalico-go/lib/options"
+	"github.com/projectcalico/libcalico-go/lib/watch"
+
+	aapi "github.com/projectcalico/api/pkg/apis/projectcalico/v3"
+)
+
+// NewCalicoNodeStatusStorage creates a new libcalico-based storage.Interface implementation for CalicoNodeStatus
+func NewCalicoNodeStatusStorage(opts Options) (registry.DryRunnableStorage, factory.DestroyFunc) {
+	c := CreateClientFromConfig()
+	getFn := func(ctx context.Context, c clientv3.Interface, ns string, name string, opts clientOpts) (resourceObject, error) {
+		ogo := opts.(options.GetOptions)
+		return c.CalicoNodeStatus().Get(ctx, name, ogo)
+	}
+	listFn := func(ctx context.Context, c clientv3.Interface, opts clientOpts) (resourceListObject, error) {
+		olo := opts.(options.ListOptions)
+		return c.CalicoNodeStatus().List(ctx, olo)
+	}
+	watchFn := func(ctx context.Context, c clientv3.Interface, opts clientOpts) (watch.Interface, error) {
+		olo := opts.(options.ListOptions)
+		return c.CalicoNodeStatus().Watch(ctx, olo)
+	}
+	dryRunnableStorage := registry.DryRunnableStorage{Storage: &resourceStore{
+		client:            c,
+		codec:             opts.RESTOptions.StorageConfig.Codec,
+		versioner:         etcd.APIObjectVersioner{},
+		aapiType:          reflect.TypeOf(aapi.CalicoNodeStatus{}),
+		aapiListType:      reflect.TypeOf(aapi.CalicoNodeStatusList{}),
+		libCalicoType:     reflect.TypeOf(api.CalicoNodeStatus{}),
+		libCalicoListType: reflect.TypeOf(api.CalicoNodeStatusList{}),
+		isNamespaced:      false,
+		get:               getFn,
+		list:              listFn,
+		watch:             watchFn,
+		resourceName:      "CalicoNodeStatus",
+		converter:         CalicoNodeStatusConverter{},
+	}, Codec: opts.RESTOptions.StorageConfig.Codec}
+	return dryRunnableStorage, func() {}
+}
+
+type CalicoNodeStatusConverter struct {
```

Comment:
Do we need this type? Is there a difference between libcalicoObject and resourceObject?